weaver/processes/wps_package.py

```diff
@@ -2114,18 +2145,47 @@ def _handler(self, request, response):
             self.update_status("Package complete.", PACKAGE_PROGRESS_DONE, STATUS_SUCCEEDED)
         return self.response
 
-    @staticmethod
-    def make_location_input(input_type, input_definition):
+    def must_fetch(self, input_ref):
+        # type: (AnyStr) -> bool
+        """
+        Figures out if file reference should be fetched immediately for local execution.
+        If anything else than local script/docker, remote ADES/WPS process will fetch it.
+        S3 are handled here to avoid error on remote WPS not supporting it.
+
+        .. seealso::
+            - :ref:`File Reference Types`
+        """
+        if self.is_ems or self.package_type == PROCESS_WORKFLOW:
+            return False
+        app_req = self.get_application_requirement()
+        if app_req["class"] not in [CWL_REQUIREMENT_APP_BUILTIN, CWL_REQUIREMENT_APP_DOCKER]:
+            if input_ref.startswith("s3://"):
+                return True
+            return False
+        return True
```

Comment:
I think I will change this around in following PR with other comment suggestion. 
Builtin app to do S3->HTTP conversion will be the best. If process is called directly with S3, it will forward is like any other HTTP and assume the WPS knows how to handle it, which could actually be the case.